ng-dev/release/notes/release-notes.ts

```diff
@@ -50,6 +60,28 @@ export class ReleaseNotes {
     return render(changelogTemplate, await this.generateRenderContext(), {rmWhitespace: true});
   }
 
+  /** Prepends the generated release note to the CHANGELOG file. */
+  async prependEntryToChangelog() {
+    /** The fully path to the changelog file. */
+    const filePath = join(this.git.baseDir, 'CHANGELOG.md');
+    /** The changelog contents in the current changelog. */
+    let changelog = '';
+    if (existsSync(filePath)) {
+      changelog = readFileSync(filePath, {encoding: 'utf8'});
+    }
+    /** The new changelog entry to add to the changelog. */
+    const entry = await this.getChangelogEntry();
+
+    writeFileSync(filePath, `${entry}\n\n${changelog}`);
+
+    try {
+      assertValidFormatConfig(this.config);
+      await formatFiles([filePath]);
```

Comment:
In what cases do we need to run a formatter? Are we emitting potentially-broken markdown here? I'm asking because this would set a precedent for running the formatter in other places too. e.g. when updating the `package.json`.

it also seems like a bad place to do this inside the `ReleaseNotes` class? making the function perform more than it suggests (as per my interpretation). Stuff is getting more convoluted that way. Should formatter instead be run when we actually prepend with some comment on why we run the formatter afterwards (e.g. when we actually create the commit)

Reply:
Yeah, I can switch it to do that instead.

Reply:
I think the piece I am unsure of if is whether formatting should be blocking, or if formatting fails we should just notify and continue on in the process.

Reply:
I think the question would be: do you we actually need to run the formatter for the MD we generate at all? 

It's a tough decision whether this is blocking or not. it kind of depends on whether we enforce formatting or not right?

Reply:
Because we don't have a standard answer at the current moment for how MD should be formatted, my thought is just to run the formatter on all of the changes to ensure it meets whatever the criteria is for the repository we are in.

Long term we should have the same formatting requirements, and then we can just have this formatting check be part of the testing of the generation of entries.

Reply:
That sounds reasonable. I'd make it non-blocking given your points above. Can we add a TODO?